docs/kubernetes/cross-cluster.md

```diff
@@ -0,0 +1,177 @@
+        Driver:            filestore.csi.storage.gke.io
+        FSType:            ext4
+        VolumeHandle:      modeInstance/us-central1-c/pvc-f5a0d4b1-79ff-4ef1-baef-018557eafb00/vol1
+        ReadOnly:          false
+        VolumeAttributes:      ip=10.150.197.194
+                            storage.kubernetes.io/csiProvisionerIdentity=1607979241304-8081-filestore.csi.storage.gke.io
+                            volume=vol1
+    ```
+
+3. Switch context to the second cluster `ClusterB` and create a pre-provisioned PersistentVolume resource pointing to the filestore instance
+    ```console
+    $ kubectl apply -f examples/kubernetes/cross-cluster/demo-deployment-clusterB.yaml
+    deployment.apps/web-server-deployment-cluster-b created
+    persistentvolumeclaim/test-pvc-fs created
+    ```
+
+    ```console
+    $ kubectl get pvc
+    NAME          STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS    AGE
+    test-pvc-fs   Bound    pre-pv   1Ti        RWX            csi-filestore   3s
+    ```
+
+    ```console
+    $ kubectl get pod
+    NAME                                     READY   STATUS    RESTARTS   AGE
+    web-server-deployment-cluster-b-6fc54d4776-d68mg   1/1     Running   0          6s
+    web-server-deployment-cluster-b-6fc54d4776-hd5x2   1/1     Running   0          7s
+    web-server-deployment-cluster-b-6fc54d4776-pjt2r   1/1     Running   0          7s
+    ```
+
+4. Ensure writes work across clusters.
+    From `ClusterB`,
+    ```console
+    $ kubectl exec web-server-deployment-cluster-b-6fc54d4776-d68mg -- touch /usr/share/nginx/html/testfile
+    $ kubectl exec web-server-deployment-cluster-b-6fc54d4776-d68mg -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+
+    $ kubectl exec web-server-deployment-cluster-b-6fc54d4776-hd5x2 -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+
+    $ kubectl exec web-server-deployment-cluster-b-6fc54d4776-pjt2r -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+    ```
+
+    Switch context to `ClusterA`. Check that the deployments in the `ClusterA`, pointing to the same filestore instance, can access the newly created file.
+    ```console
+    $ kubectl exec web-server-deployment-cluster-a-6fc54d4776-kssrl -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+
+    $ kubectl exec web-server-deployment-cluster-a-6fc54d4776-lw2l2 -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+
+    $ kubectl exec web-server-deployment-cluster-a-6fc54d4776-sxrnd -- ls /usr/share/nginx/html
+    lost+found
+    testfile
+    ```
+
+5. Deletion of workloads and the volume resources.
+
+    Since the same filestore instance is shared across clusters, and the workload in one cluster is unaware of the filestore instance being shared by a different cluster, care must be taken by the user to ensure that no running workload is pointing filestore isntance at the time of deletion. Deletion of the filestore instance can be initiated from any cluster.
```

Comment:
Maybe change depending on if you change the retain policy.

Reply:
done